### PR TITLE
sign up button and logic for events

### PR DESCRIPTION
**Pull request by Jeppes**
Added ability to sign up for events

### CODE REVIEW

File: app/src/main/java/se/ths/kth/Aramda/components/events/detail/EventDetailFragment.java [hunk 1]
```diff
@@ -117,6 +123,43 @@ public void setEvent(Event event) {
...
+    private void setupNoEventSignup(TextView eventStatus) {
+        String noSignupMessage = getContext().getString(R.string.no_signup_message);
+        eventStatus.setText(noSignupMessage);
+    }
+
+    private void setupEventButton(Event event, Button button) {
+        Date today = new Date();
+
+        button.setVisibility(View.VISIBLE);
+        if (event.getSignup_starts_at().compareTo(today) <= 0 && event.getSignup_ends_at().compareTo(today) > 0) {
```
Comment (apals) [inline app/src/main/java/se/ths/kth/Aramda/components/events/detail/EventDetailFragment.java]: can use `date.before(d)` instead

File: app/src/main/java/se/ths/kth/Aramda/utils/Utils.java [hunk 2]
```diff
@@ -68,4 +71,17 @@ public static Spanned fromHtml(String description) {
...
+    public static int getColor(Context context, int id) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            return context.getColor(id);
+        } else {
+            return context.getResources().getColor(id);
+        }
+    }
+
+    public static void openLink(Context context, String url) {
+        Intent intent= new Intent(Intent.ACTION_VIEW, Uri.parse(url));
```
Comment (apals) [inline app/src/main/java/se/ths/kth/Aramda/utils/Utils.java]: `intent= new` :( please whitespace before `=`

Comment (Jeppes) [inline app/src/main/java/se/ths/kth/Aramda/utils/Utils.java]: true

File: app/src/main/java/se/ths/kth/Aramda/components/events/detail/EventDetailFragment.java [hunk 3]
```diff
@@ -117,6 +123,43 @@ public void setEvent(Event event) {
...
         binding.content.eventDescriptionTextview.setText(Utils.fromHtml(event.getDescription()));
         binding.content.eventDescriptionTextview.setMovementMethod(LinkMovementMethod.getInstance());
+
+        setEventStatus(event, binding);
+    }
+
+    private void setEventStatus(Event event, FragmentEventDetailBinding binding) {
+        if (!event.getRegistration_required() || event.getSignup_starts_at() == null) {
+            setupNoEventSignup(binding.content.eventStatusTextview);
+            binding.content.signUpButton.setVisibility(View.GONE);
```
Comment (Jeppes) [inline app/src/main/java/se/ths/kth/Aramda/components/events/detail/EventDetailFragment.java]: maybe, i'll add it for good measure

**Comment by Jeppes** [comment]
@apals fixed

**Pull request merged**